samples/wifi/raw_tx_packet/src/main.c

```diff
@@ -0,0 +1,485 @@
+
+	if (net_mgmt(NET_REQUEST_WIFI_CONNECT, iface,
+		     &cnx_params, sizeof(struct wifi_connect_req_params))) {
+		LOG_ERR("Connection request failed");
+
+		return -ENOEXEC;
+	}
+
+	LOG_INF("Connection requested");
+
+	return 0;
+}
+#endif
+
+#ifdef CONFIG_RAW_TX_PACKET_SAMPLE_IDLE_MODE
+static void wifi_set_channel(void)
+{
+	struct net_if *iface;
+	struct wifi_channel_info channel_info = {0};
+	int ret;
+
+	channel_info.oper = WIFI_MGMT_SET;
+
+	if (channel_info.if_index == 0) {
+		iface = net_if_get_first_wifi();
+		if (iface == NULL) {
+			LOG_ERR("Cannot find the default wifi interface");
+			return;
+		}
+		channel_info.if_index = net_if_get_by_iface(iface);
+	} else {
+		iface = net_if_get_by_index(channel_info.if_index);
+		if (iface == NULL) {
+			LOG_ERR("Cannot find interface for if_index %d",
+					      channel_info.if_index);
+			return;
+		}
+	}
+
+	if (channel_info.oper == WIFI_MGMT_SET) {
+		channel_info.channel = CONFIG_RAW_TX_PACKET_SAMPLE_CHANNEL;
+		if ((channel_info.channel < WIFI_CHANNEL_MIN) ||
+			   (channel_info.channel > WIFI_CHANNEL_MAX)) {
+			LOG_ERR("Invalid channel number. Range is (1-233)");
+			return;
+		}
+	}
+
+	ret = net_mgmt(NET_REQUEST_WIFI_CHANNEL, iface,
+			&channel_info, sizeof(channel_info));
+
+	if (ret) {
+		LOG_ERR(" Channel setting failed %d\n", ret);
+			return;
+	}
+
+	LOG_INF("Wi-Fi channel set to %d", channel_info.channel);
+}
+#endif
+
+static void wifi_set_mode(void)
+{
+	int ret;
+	struct net_if *iface;
+	struct wifi_mode_info mode_info = {0};
+
+	mode_info.oper = WIFI_MGMT_SET;
+
+	if (mode_info.if_index == 0) {
+		iface = net_if_get_default();
+		if (iface == NULL) {
+			LOG_ERR("Cannot find the default wifi interface");
+			return;
+		}
+		mode_info.if_index = net_if_get_by_iface(iface);
+	} else {
+		iface = net_if_get_by_index(mode_info.if_index);
+		if (iface == NULL) {
+			LOG_ERR("Cannot find interface for if_index %d",
+				      mode_info.if_index);
+			return;
+		}
+	}
+
+	mode_info.mode =  WIFI_STA_MODE | WIFI_TX_INJECTION_MODE;
+
+	ret = net_mgmt(NET_REQUEST_WIFI_MODE, iface, &mode_info, sizeof(mode_info));
+	if (ret) {
+		LOG_ERR("Mode setting failed %d", ret);
+	}
+}
+
+static void wifi_send_raw_tx_packet(void)
+{
+	struct sockaddr_ll sa;
+	int sockfd, ret;
+	struct net_if *iface;
+	struct nrf_wifi_fmac_rawpkt_info raw_tx_pkt;
+	char *buffer = NULL;
+	int buf_length, transmission_mode, num_frames;
+
+	/* Create a raw socket */
+	sockfd = socket(AF_PACKET, SOCK_RAW, IPPROTO_RAW);
+	if (sockfd < 0) {
+		LOG_ERR("Unable to create a socket");
+		return;
+	}
+
+	iface = net_if_get_default();
+	if (!iface) {
+		LOG_ERR("Unable to get default interface");
+		return;
+	}
+
+	sa.sll_family = AF_PACKET;
+	sa.sll_ifindex = net_if_get_by_iface(iface);
+
+	/* Bind the socket */
+	ret = bind(sockfd, (struct sockaddr *)&sa, sizeof(struct sockaddr_ll));
+	if (ret < 0) {
+		LOG_ERR("Error: Unable to bind socket to the network interface:%d", errno);
+		close(sockfd);
+		return;
+	}
```

Comment:
This can be a helper function to `setup_raw_socket()` and outside `send_raw_tx` as it can be called many times.